spec/02-integration/11-dbless/03-config_persistence_spec.lua

```diff
@@ -124,31 +124,38 @@ describe("dbless persistence with a declarative config #off", function()
         database = "off",
         declarative_config = yaml_file,
     }))
-    local proxy_client = assert(helpers.proxy_client())
-    local res = assert(proxy_client:get("/test", { headers = { host = "example1.dev" } }))
-    assert.res_status(401, res)
 
-    res = assert(proxy_client:get("/500", { headers = { host = "example1.dev" } }))
-    assert.res_status(404, res) -- 404, only the declarative config is loaded
-    proxy_client:close()
+    assert
+      .with_timeout(15)
+      .eventually(function ()
+        local proxy_client = assert(helpers.proxy_client())
+        local res = assert(proxy_client:get("/test", { headers = { host = "example1.dev" } }))
+        assert.res_status(401, res) -- 401, should load the declarative config
+
+        res = assert(proxy_client:get("/500", { headers = { host = "example1.dev" } }))
+        assert.res_status(404, res) -- 404, should not load the persisted lmdb config
+
+        proxy_client:close()
+      end)
```

Comment:
This is missing the predicate/condition for `eventually()`. It needs one of `is_truthy`/`is_falsy`/`has_error`/`has_no_error`, otherwise this code will never run.

https://github.com/Kong/kong/blob/985119016f67872b9d9d2d7140be969a994a6cc7/spec/helpers/wait.lua#L451-L466

Reply:
done